zebra/zebra_srv6_vty.c

```diff
@@ -941,11 +998,150 @@ DEFPY(no_srv6_sid_format_explicit,
+		sidaction = ZEBRA_SEG6_LOCAL_ACTION_END_UDT6;
+		vrfName = argv[idx + 2]->arg;
+	} else if (argv_find(argv, argc, "uDT46", &idx)) {
+		sidaction = ZEBRA_SEG6_LOCAL_ACTION_END_UDT46;
+		vrfName = argv[idx + 2]->arg;
+	}
+	prefix = argv[1]->arg;
+	ret = str2prefix_ipv6(prefix, &ipv6prefix);
+	if (!ret) {
+		vty_out(vty, "Malformed IPv6 prefix\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+	if (!prefix_match((struct prefix *)&locator->prefix, (struct prefix *)&ipv6prefix)) {
+		vty_out(vty, "SID prefix must match locator prefix\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+	if (vrfName) {
+		vrf = zebra_vrf_lookup_by_name(vrfName);
+		if (!vrf) {
+			vty_out(vty, "Vrf does not exist\n");
+			return CMD_WARNING_CONFIG_FAILED;
+		}
+	}
+	for (ALL_LIST_ELEMENTS_RO(locator->sids, node, srv6_sid)) {
+		if (IPV6_ADDR_SAME(&srv6_sid->ipv6Addr.prefix, &ipv6prefix.prefix)) {
+			vty_out(vty, "Prefix %s is already exist, please delete it first.\n",
+				argv[1]->arg);
+			return CMD_WARNING;
+		}
+	}
+	srv6_sid = sid_lookup_by_vrf_action(locator, vrfName, sidaction);
+	if (srv6_sid) {
+		vty_out(vty, "VRF %s is already exist, please delete it first.\n", vrfName);
+		return CMD_WARNING;
+	}
+	srv6_sid = srv6_locator_sid_alloc();
+	srv6_sid->sidaction = sidaction;
+	if (vrfName != NULL)
+		strlcpy(srv6_sid->vrfName, vrfName, VRF_NAMSIZ);
+	else
+		strlcpy(srv6_sid->vrfName, VRF_DEFAULT_NAME, VRF_NAMSIZ);
+
+	srv6_sid->ipv6Addr = ipv6prefix;
+	strlcpy(srv6_sid->sidstr, prefix, PREFIX_STRLEN);
+
+	listnode_add(locator->sids, srv6_sid);
+
+	for (ALL_LIST_ELEMENTS_RO(zrouter.client_list, client_node, client))
+		zsend_srv6_manager_get_locator_sid_response(client, VRF_DEFAULT, locator);
+	return CMD_SUCCESS;
+}
+
+DEFPY(no_locator_sid,
+	  no_locator_sid_cmd,
+      "no sid WORD behavior <uN | uDT4 vrf VIEWVRFNAME | uDT6 vrf VIEWVRFNAME | uDT46 vrf VIEWVRFNAME>",
+      NO_STR
+      "Configure SRv6 locator sid\n"
+      "Specify SRv6 locator hex sid\n"
+      "Configure behavior\n"
+      "Apply the code to an End.uN SID\n"
+      "Apply the code to an End.uDT4 SID\n"
+      "vrf\n"
+      "vrf\n"
+      "Apply the code to an End.uDT6 SID\n"
+      "vrf\n"
+      "vrf\n"
+      "Apply the code to an End.uDT46 SID\n"
+      "vrf\n"
+      "vrf\n")
+{
+	VTY_DECLVAR_CONTEXT(srv6_locator, locator);
+	struct seg6_sid *srv6_sid = NULL;
+	struct listnode *node, *next;
+	char *prefix = NULL;
+	int ret = 0;
+	struct prefix_ipv6 ipv6prefix = { 0 };
+	struct zserv *client;
+	struct listnode *client_node;
+
+	prefix = argv[2]->arg;
+	ret = str2prefix_ipv6(prefix, &ipv6prefix);
+	if (!ret) {
+		vty_out(vty, "Malformed IPv6 prefix\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+
+	for (ALL_LIST_ELEMENTS(locator->sids, node, next, srv6_sid)) {
+		if (IPV6_ADDR_SAME(&srv6_sid->ipv6Addr.prefix, &ipv6prefix.prefix)) {
+			for (ALL_LIST_ELEMENTS_RO(zrouter.client_list, client_node, client))
+				zsend_srv6_manager_del_sid(client, VRF_DEFAULT, locator, srv6_sid);
```

Comment:
- A registration scheme would avoid sending unnecessary messages to every daemon.
For instance, the sending could be triggered by some config in BGP, ISIS, PATHD. Eventually, this should be only sent to those daemons.

- Why sending all the sids while only one is configured ?